Update Readme (#125)
amzn/ion-go: README.md

[![Build Status](https://github.com/amzn/ion-go/workflows/Go%20Build/badge.svg)](https://github.com/amzn/ion-go/actions?query=workflow%3A%22Go+Build%22)

This package is for parsing and writing text and binary based Ion data.
Amazon Ion ( http://amzn.github.io/ion-docs/ ) library for Go

> ***This package is considered experimental, under active early development,  and the API is subject to change.***

Author: Done.

[![Build Status](https://github.com/amzn/ion-go/workflows/Go%20Build/badge.svg)](https://github.com/amzn/ion-go/actions?query=workflow%3A%22Go+Build%22)

Amazon Ion ( http://amzn.github.io/ion-docs/ ) library for Go

***This package is consider beta. While the API is relatively stable it is still subject to change***

This package is based on work from David Murray ([fernomac](https://github.com/fernomac/)) on https://github.com/fernomac/ion-go.

> The Ion team greatly appreciates David's contributions to the Ion community.

Reviewer: 😊

(Thanks again for picking this up and running with it, I'm excited to have an officially-supported version!)

}
```

In order to Marshal/Unamrshal Ion values with annotation, we use a Go struct of type []stirng and tag it with
`ion:",annotation"`. The other field would be what our Ion value can fit in (both in terms of type and size). For instance,
to Marshal `age::20`, it must be in a struct as below:
```GO
  type foo struct {
    Value   interface{}
    AnyName []string `ion:",annotations"`
  }
  data := foo{20, []string{"age"}}

>   val, err := MarshalText(data)

Reviewer 2: ```suggestion
  val, err := ion.MarshalText(data)
```

  if err != nil {

>      t.Fatal(err)

Reviewer 2: The goal is to have, as close as possible, copy paste code that works. `t.Fatal` seems like a leftover from a copy paste from our tests :)

Author: Ah! There is always one which gets away. Thanks for noticing it.

>      t.Fatal(err)

Reviewer 2: ```suggestion
     panic(err)
```

>   }

Reviewer 2: Maybe add the line that would print out the Ion value as a string and the expected result and comment it out so that the reader would have an idea of the expected output.

Author: Thanks for the suggestion. Added.

>   }

Reviewer 2: ```suggestion
  fmt.Println("Ion text: ", string(val)) // Ion text: age::20
  }
```

```

And to Unmarshal the same data, follwing struct can be used:
```Go
  type foo struct {
    Value   interface{}
    AnyName []string `ion:",annotations"`
  }
  var val foo

>   err := UnmarshalString("age::20", &val)

Reviewer 2: ```suggestion
  err := ion.UnmarshalString("age::20", &val)
```

  if err != nil {

>     t.Fatal(err)

Reviewer 2: ```suggestion
    panic(err)
```

>   }

Reviewer 2: ```suggestion
    fmt.Println("Val = ", val) // Val =  {20 [age]}
  }
```

### Reading and Writing

For low-level streaming read and write access, use a `Reader` or `Writer`.

> Following example shows how to create a reader, reading values from that reader,

Reviewer 2: ```suggestion
The following example shows how to create a reader, read values from that reader,
```

> and writing them in a writer:

Reviewer 2: ```suggestion
and write those values out using a writer:
```

  }
}
```
### License

> This library is licensed under the MIT License.

Reviewer: Apache 2.0? (I'm not _necessarily_ opposed to relicensing my contributions, but if you are going to change the license I'd be interested to hear more about why.)

Reviewer 2: Ha, good catch. There is no re-licensing. This is a copy paste error :)

### License

> This library is licensed under the MIT License.

Reviewer 2: ```suggestion
This library is licensed under the Apache 2.0 License.
```
